src/core/program.rs

```diff
@@ -429,6 +429,7 @@ impl Program {
         self.context.set_render_states(render_states);
         self.use_program();
         unsafe {
+            self.context.bind_vertex_array(Some(self.context.vao));
```

Comment:
I guess this could be moved to `full_screen_draw`? It makes it a bit more explicit why we need it in that case.

Reply:
Yeah, I wasn't super sure about this. I guess full screen is one of the only cases where we're likely to have no geometry - I'll move it

Reply:
I'm also not entirely sure. I guess in principle, there should be a method on Program that binds the VAO in case you don't want to bind any vertex attributes. Or maybe automatically before a draw call as you did. But well, I'll rather wait and see if that's going to be needed before I want to add that.